iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/utils/CommandDetector.java

```diff
@@ -63,20 +66,69 @@ public static Builder builder() {
    * The foremost element is taken from the stack and checked to see if it matches the command to be searched for.
    */
   public List<Command> search(List<ArgumentResolution> resolvedArguments) {
+    SeparatedList<List<ArgumentResolution>, String> splitCommands = splitCommands(resolvedArguments);
     List<Command> commands = new ArrayList<>();
-    Deque<ArgumentResolution> argumentStack = new LinkedList<>(resolvedArguments);
 
-    PredicateContext context = new PredicateContext(argumentStack, predicates);
+    for (List<ArgumentResolution> resolved : splitCommands.elements()) {
+      Deque<ArgumentResolution> argumentStack = new LinkedList<>(resolved);
 
-    while (!argumentStack.isEmpty()) {
-      List<ArgumentResolution> commandArguments = fullMatch(context);
-      if (!commandArguments.isEmpty()) {
-        commands.add(new Command(commandArguments));
+      PredicateContext context = new PredicateContext(argumentStack, predicates);
+
+      while (!argumentStack.isEmpty()) {
+        List<ArgumentResolution> commandArguments = fullMatch(context);
+        if (!commandArguments.isEmpty()) {
+          commands.add(new Command(commandArguments));
+        }
       }
     }
     return commands;
   }
 
+  private static final String COMMAND_WITHOUT_OPERATOR = "(?:\"(?:\\\\.|[^\"])*+\"|'(?:\\\\.|[^'])*+'|[^;&|])*+";
+  private static final String OPERATOR_REGEX = ";|&&|&|\\|\\||\\|";
+  private static final Pattern PATTERN_MULTIPLE_OPERATOR_DETECTION_BEFORE = Pattern.compile(
+    "(?<before>" + COMMAND_WITHOUT_OPERATOR + ")(?<operator>" + OPERATOR_REGEX + ")");
+
+  // Cognitive Complexity of methods should not be too high
+  @SuppressWarnings("java:S3776")
+  private static SeparatedList<List<ArgumentResolution>, String> splitCommands(List<ArgumentResolution> resolvedArguments) {
```

Comment:
We agreed that the method is too complex and will be improved.
However, the idea of moving this logic to `CheckUtils` seems to be not perfect - before each call of the `CommandDetector. search()` method the moved one needs to be called too.

Reply:
We decided to keep it in `CommandDetector` and make it public, and have 2 versions of the search method, because it will be needed in checks that verify something in the first or last command.

Reply:
To be honest I'm still not happy that this functionality resides directly in `CommandDetector`. I can understand not to put it in CheckUtils, but with respect of separation of concerns I would use a nested class or newly created utility class that only the detector uses.